### ISSUE TITLE
Switch from doubles to floats

### ISSUE THREAD

**Issue by aburt2** [malloch/IMU_Sensor_Fusion#1]
@malloch Would it cause any problems to switch from doubles to floats for the sensor fusion? For SoCs like the RW612 and the ESP32 whose FPU only supports single bit float operation it would significantly improve performance. 

For the RW612 it halved the update time from ~0.5ms to ~0.2ms. 

Basically make the changes I made here: https://github.com/aburt2/IMU_Sensor_Fusion/tree/zephyr

**Comment by aburt2** [comment]
Here is some better data. Tested by doing 100,000 iterations of the following calculation

```
      sensor_fusion.setAccelerometerValues(0, 1, 0);
      sensor_fusion.setGyroscopeRadianValues(0, 0, 0, 0.001);
      sensor_fusion.setMagnetometerValues(0, 1, 2);
      sensor_fusion.update();
```

TLDR: Using floats can offer up to a 10 times performance increase

ESP32S3 @ 240MHz, 100,000 iterations
| Conditions    | Total time (ms) | Unit time (us) |
| -------------- | ---------------- | ---------------|
| __Doubles__  | 29914ms          | 299.1us          |
| __Floats__      | 2627ms            | 26.3us            |

IrisW106 (RW612 SoC) @ 260MHz, 100,000 iterations
| Conditions    | Total time (ms) | Unit time (us) |
| -------------- | ---------------- | ---------------|
| __Doubles__  | 14468ms          | 144.7us          |
| __Floats__      | 1375ms            | 13.7us            |

**Comment by aburt2** [comment]
@malloch added a pull request, to support the use of floats if speed is preferred over accuracy

**Comment by malloch** [comment]
Merged! Thanks for all the profiling and patching work!
